system/jlib/jfcmp.hpp

```diff
@@ -0,0 +1,516 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jlzw.hpp"
+
+#define COMMITTED ((size32_t)-1)
+
+#define FCMP_BUFFER_SIZE (0x100000)
+
+class jlib_decl CFcmpCompressor : public CSimpleInterfaceOf<ICompressor>
```

Comment:
minor: we used to use .ipp's to hide away implementation detail classes/methods shared between modules in the same shared library that would never be exposed to others, i.e. that others should not include or rely on staying static.

We avoid introducing new .ipp's now.
Was discussing with @richardkchapman earlier, in cases like this it may make sense to using a naming convention for the header to indicate that it is internal only, e.g. jcomp_internal.hpp. In which case it wouldn't need the jlib_decl either.

Wouldn't change now though, something for us to think about.